project/api/models.py

```diff
@@ -87,9 +90,16 @@ def clean(self, *args, **kwargs):
         if self.start_date > self.end_date:
             raise ValidationError("Start date cannot be later than end date")
         if loads(self.geo.json)["type"] != "Polygon" and loads(self.geo.json)["type"] != "MultiPolygon":
-            raise ValidationError("Only Polygon and MultiPolygon objects are acceptable geometry types")
+            raise ValidationError(
+                "Only Polygon and MultiPolygon objects are acceptable geometry types")
         super(Territory, self).clean(*args, **kwargs)
 
+        territories = Territory.objects.filter(nation=self.nation)
+        for territory in territories:
```

Comment:
Consider replacing this for loop with more efficient db queries (https://docs.djangoproject.com/en/2.1/ref/models/querysets/#range) if it's not too hard.

Reply:
Sure thing

Reply:
Well I'm not sure if it's worth it but I did think this might create problems in the long run but decided to keep it simple and wait for a feedback like this to change it. I guess it can't hurt.